### ISSUE TITLE
Provide an example for transforming a Stream

### ISSUE THREAD

**Issue by iLoveDocs** [dart-lang/site-www#4322]
### What information is missing?

The [CONSIDER using higher-order methods to transform a stream.](https://dart.dev/guides/language/effective-dart/usage#consider-using-higher-order-methods-to-transform-a-stream) section mentions: 

> Streams support many of the same methods and also handle things like transmitting errors, closing, etc. correctly.

It would be nice to provide some examples. 

### How would you like us to fix this problem?

An example could be: 

```dart
var intStream = Stream.fromIterable([1, 2, 3]); // Stream<int>
var boolStream = intStream.map((n) => n.isEven); // Stream<bool>
```

**Comment by MaryaBelanger** [comment]
This page is probably a little out of date, but might help? https://dart.dev/articles/libraries/creating-streams
(I didn't actually check myself, so let us know if it's lacking)

**Comment by parlough** [comment]
I agree, it would be nice to expand on this section, rather than just pointing to another one, so that the rule can be absorbed in isolation.
